neat_core.c

```diff
@@ -804,6 +1035,210 @@ neat_open(neat_ctx *mgr, neat_flow *flow, const char *name, uint16_t port)
     return neat_he_lookup(mgr, flow, he_connected_cb);
 }
 
+neat_error_code
+neat_open_localname(neat_ctx *mgr, neat_flow *flow, const char *name, uint16_t port, const char *localname)
+{
+    neat_log(NEAT_LOG_DEBUG, "%s", __func__);
+
+    if (flow->name) {
+        return NEAT_ERROR_BAD_ARGUMENT;
+    }
+
+    if (localname == NULL)
+	return neat_open(mgr, flow, name, port);
+
+    return NEAT_ERROR_UNABLE;
+}
+
+neat_error_code
+neat_open_multistream(neat_ctx *mgr, neat_flow *flow, const char *name, uint16_t port, const char* localname, int count)
+{
+    neat_log(NEAT_LOG_DEBUG, "%s", __func__);
+
+    if (count < 1) {
+        return NEAT_ERROR_BAD_ARGUMENT;
+    } else if (count > 1) {
+	// TODO: multi-streaming is not yet available from the SCTP
+	// layer.
+	return NEAT_ERROR_UNABLE;
+    }
+
+    return neat_open_localname(mgr, flow, name, port, localname);
+}
+
+neat_error_code
+neat_change_timeout(neat_ctx *mgr, neat_flow *flow, int seconds)
+{
+    neat_log(NEAT_LOG_DEBUG, "%s", __func__);
+
+    // TCP User Timeout isn't supported by these platforms:
+#if !(defined(__FreeBSD__) || defined(__NetBSD__) || defined (__APPLE__))
+    if (seconds < 0) {
+            neat_log(NEAT_LOG_WARNING,
+                    "Unable to change timeout: "
+                    "Negative timeout specified");
+
+            return NEAT_ERROR_BAD_ARGUMENT;
+    }
+
```

Comment:
Why is this wrong only on the specified platforms? Wouldn't it make sense to use an unsigned integer as the type for `seconds`?